docs/resources/restore.md

```diff
@@ -0,0 +1,102 @@
+
+## Example Usage
+
+```terraform
+resource "cockroach_restore" "database_restore" {
+  destination_cluster_id = "12345678-1234-4123-8123-123456789abc"
+  type                   = "DATABASE"
+  source_cluster_id      = "deadbeef-dead-4bee-8def-deadbeefcafe"
+
+  restore_opts = {
+    new_db_name = "new_database"
+    schema_only = true
+  }
+
+  object {
+    database = "database1"
+  }
+}
+
+resource "cockroach_restore" "table_restore" {
+  destination_cluster_id = "12345678-1234-4123-8123-123456789abc"
+  type                   = "TABLE"
+  backup_id              = "11111111-1111-4111-8111-111111111111"
+
+  restore_opts = {
+    into_db                   = "other_database"
+    skip_localities_check     = true
+    skip_missing_foreign_keys = true
+    skip_missing_sequences    = true
+  }
+
+  object {
+    database = "database1"
+    schema   = "schema1"
+    table    = ["table_a", "table_b"]
+  }
+
+  object {
+    database = "database2"
+    schema   = "schema2"
+    table    = ["table_c"]
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `destination_cluster_id` (String) ID of the cluster where the backup will be restored.
+- `type` (String) The type of restore to be performed. Allowed values are: `CLUSTER` or `DATABASE` or `TABLE`.
+
+### Optional
+
+- `backup_id` (String) ID of the managed backup from which data will be restored. If this value is not set, the restore operation uses the most recent available backup on the source cluster.
+- `objects` (Block List) The list of objects to restore. Required when the restore operation type is DATABASE or TABLE. (see [below for nested schema](#nestedblock--objects))
+- `restore_opts` (Attributes) Additional options controlling the behavior of the restore operation. (see [below for nested schema](#nestedatt--restore_opts))
+- `source_cluster_id` (String) ID of the source cluster containing the managed backup to be restored.
+
+### Read-Only
+
+- `completion_percent` (Number) Decimal value showing the percentage of the restore operation that has been completed. Value ranges from 0 to 1.
+- `created_at` (String) The time at which the restore operation was initiated.
+- `id` (String) The ID of this resource.
+- `status` (String) The current status of the restore operation.
+
+<a id="nestedblock--objects"></a>
+### Nested Schema for `objects`
+
+Required:
+
+- `database` (String) The database name in the fully qualified name of the objects to be restored. In the case of an operation that restores `tpcc.public.warehouse`, this value would be `tpcc`.
+
+Optional:
+
+- `schema` (String) The schema name in the fully qualified name of the objects to be restored. In the case of an operation that restores `tpcc.public.warehouse`, this value would be `public`.
+- `tables` (List of String) The table name in the fully qualified name of the objects to be restored. In the case of an operation that restores `tpcc.public.warehouse`, this would value be `warehouse`.
```

Comment:
```suggestion
- `schema` (String) The schema name in the fully qualified name of the objects to be restored. In the case of a job that restores `tpcc.public.warehouse`, this value would be `public`.
- `tables` (List of String) The table name in the fully qualified name of the objects to be restored. In the case of a job that restores `tpcc.public.warehouse`, this would value be `warehouse`.
```